### ISSUE TITLE
Support for electron

### ISSUE THREAD

**Issue by ravipati2008** [Venemo/node-lmdb#151]
I'm trying to use node-lmdb with Electron but running into an issue with `win_delay_load_hook` being set to false in `binding.gyp`. Electron needs this to be set to true in order to work on versions > 4.x. 
https://github.com/electron/electron/blob/master/docs/tutorial/using-native-node-modules.md#a-note-about-win_delay_load_hook
It would also be nice if pre-built binaries are available for Electron so we don't have to run into these build issues in the first place.

**Comment by ravipati2008** [comment]
Turns out this is an intentional change in order for builds to work on windows.
https://github.com/Venemo/node-lmdb/commit/32d1ee6f21f86f7a65dfcee7ca0c2de11717adc1

I can confirm that I'm running into errors when building with `node-gyp` with `win_delay_load_hook` set to true but I'm able to get it working using another package called `electron-rebuild` which builds specifically for `Electron`.

**Comment by Venemo** [comment]
Hi,
I don't use Electron so I'm not sure what needs to be done. Pull requests are welcome. If you know how to get it working then I'm also happy to help and/or add the necessary support for Electron.

**Comment by ravipati2008** [comment]
If we can get node-lmdb compiling with `win_delay_load_hook` set to true in `binding.gyp`, that should be good enough to support Electron.
Right now, I'm getting compilation errors when I set this to true and have no clue as to how to fix this:
```
$ node-gyp build
gyp info it worked if it ends with ok
gyp info using node-gyp@5.0.3
gyp info using node@10.13.0 | win32 | x64
gyp info spawn C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\MSBuild\15.0\Bin\MSBuild.exe
gyp info spawn args [ 'build/binding.sln',
gyp info spawn args   '/clp:Verbosity=minimal',
gyp info spawn args   '/nologo',
gyp info spawn args   '/p:Configuration=Release;Platform=x64' ]
Building the projects in this solution one at a time. To enable parallel build, please add the "/m" switch.
  mdb.c
  midl.c
node-lmdb\dependencies\lmdb\libraries\liblmdb\mdb.c(4334): warning C4996: 'GetVersion': was declared deprecated [node-lmdb\build\node-lmdb.vcxproj]
  c:\program files (x86)\windows kits\10\include\10.0.17763.0\um\sysinfoapi.h(135): note: see declaration of 'GetVersion'
node-lmdb\dependencies\lmdb\libraries\liblmdb\mdb.c(6898): warning C4146: unary minus operator applied to unsigned type, result still unsigned [node-lmdb\build\node-lmdb.vcxproj]
node-lmdb\dependencies\lmdb\libraries\liblmdb\mdb.c(7475): warning C4333: '>>': right shift by too large amount, data loss [node-lmdb\build\node-lmdb.vcxproj]
  node-lmdb.cpp
  env.cpp
  misc.cpp
  txn.cpp
  dbi.cpp
  cursor.cpp
  win_delay_load_hook.cc
LINK : fatal error LNK1194: cannot delay-load 'node.exe' due to import of data symbol '"__declspec(dllimport) const v8::String::ExternalStringResource::`vftable'" (__imp_??_7ExternalStringResource@String@v8@@6B@)'; link without /DELAYLOAD:node.exe [node-lmdb\build\node-lmdb.vcxproj]
gyp ERR! build error
gyp ERR! stack Error: `C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\MSBuild\15.0\Bin\MSBuild.exe` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (AppData\Local\Yarn\Data\global\node_modules\node-gyp\lib\build.js:196:23)
gyp ERR! stack     at ChildProcess.emit (events.js:182:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:240:12)
gyp ERR! System Windows_NT 10.0.18362
gyp ERR! command "C:\\nodejs\\node.exe" "AppData\\Local\\Yarn\\Data\\global\\node_modules\\node-gyp\\bin\\node-gyp.js" "build"
gyp ERR! cwd node-lmdb
gyp ERR! node -v v10.13.0
gyp ERR! node-gyp -v v5.0.3
gyp ERR! not ok

```

**Comment by Venemo** [comment]
What exactly does `electron-rebuild` do?

**Comment by ravipati2008** [comment]
Made some progress here. I'm able to get it working with `node-gyp --debug`. This is also the reason `electron-rebuild` was working. It's just executing `node-gyp rebuild` with `--debug` which I was passing into `electron-rebuild` inadvertently.  Without that flag electron-rebuild also fails. So, we can ignore that package.
I'm still trying to figure out why it works in debug mode but not in release mode. In release mode, compilation is fine and this error happens during link time.
Could be related to this code but not really sure:
```
src/node-lmdb.h
678:class CustomExternalStringResource : public String::ExternalStringResource {
```

**Comment by Venemo** [comment]
Yes, the error definitely hints at `ExternalStringResource` but I don't know why. Also the `ExternalStringResource` is an essential part of node-lmdb and I can't just remove it for electron.

**Comment by Venemo** [comment]
If you figure it out, feel free to send me a pull request.

**Comment by d4tocchini** [comment]
@ravipati2008 @Venemo Here's how I setup proper electron support: https://github.com/kriszyp/node-lmdb/pull/1

I've only tested on MacOS, but should work for general electron usage.

**Comment by d4tocchini** [comment]
If you don't care for context-aware support, you can build for electron with something like:

```sh
HOME=~/.electron-gyp node-gyp rebuild  --target=9.0.0 --runtime=electron --arch=x64 --dist-url=https://electronjs.org/headers
```

**Comment by kriszyp** [comment]
I think this has already been addressed, but 0.9.4 is definitely tested with context-aware modules and even does some prebuilds for electron. Let me know if you see any issues.
